rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java

```diff
@@ -350,6 +350,13 @@ NodeSet getNodes(int number, TopologyDescriptor descriptor, List<SelectionScript
      */
     NodeSet getNodes(Criteria criteria);
 
+    /**
+     *
+     * @param nodeSourceName
+     * @return true if RM already has node source with given name
+     */
+    boolean isNodeSourceAlreadyExisting(String nodeSourceName);
```

Comment:
Better to use BooleanWrapper to let the caller decide to handle it synchronously or not.

Reply:
yes right, make sense